friendships/services.py

```diff
@@ -38,4 +43,25 @@ def has_followed(cls, from_user, to_user):
         return Friendship.objects.filter(
             from_user=from_user,
             to_user=to_user,
-        ).exists()
\ No newline at end of file
+        ).exists()
+
+    @classmethod
+    def get_following_user_id_set(cls, from_user_id):
+        # LRU
+        key = FOLLOWINGS_PATTERN.format(user_id=from_user_id)
+        user_id_set = cache.get(key)
+        if user_id_set is not None:
+            return user_id_set
+        # get from the db
+        friendships = Friendship.objects.filter(from_user_id=from_user_id)
+        user_id_set = set([
+            fs.to_user_id
+            for fs in friendships
+        ])
+        cache.set(key, user_id_set)
+        return user_id_set
+
+    @classmethod
+    def invalidate_following_cache(cls, from_user_id):
```

Comment:
the right way to keep cache to have updated data is to invalidate it  in some frequence